python/tchannel/tornado/peer.py

```diff
@@ -169,13 +179,22 @@ def request(self, **kwargs):
         :param service_threshold:
             If ``hostport`` was not specified, this specifies the score
             threshold at or below which peers will be ignored.
+        :param black_list:
+            Peers on the black_list won't be chosen.
         """
-        return self.choose(
+        kwargs['peer_group'] = self
```

Comment:
Additionally it's generally a bad idea to pass around `self`. This easily leaks memory and blurs class responsibilities.